### ISSUE TITLE
iOS 7 support

### ISSUE THREAD

**Issue by elanini** [jlippold/WifiSMS#86]
This is currently lacking iOS 7 support. I would enjoy using this if support were added. Thank you.

**Comment by chid** [comment]
I doubt there is much interest now that there's http://remotemessages.com/ released for iOS 7.

**Comment by akohlsmith** [comment]
There is considerable interest in an open source SMS gateway. The ability to tweak and patch the source is a huge plus in my opinion.

**Comment by Tatsh** [comment]
I agree with you @akohlsmith but it's a big undertaking to do for free by one person. So I just pay for Remote Messages for now.

**Comment by chid** [comment]
the ideal of open source, unfortunately not many people contribute https://github.com/jlippold/WifiSMS/pulls

I don't know, I'm pretty happy with how responsive Beastsoft are to requests. Much of their front-end is editable, sounds, interface, shortcuts is customizable. You can even write your own custom clients using their websocket API, it's actually really neat to use.

**Comment by Tatsh** [comment]
@chid That is pretty sweet. I will consider writing something up for Pidgin so I can have one less tab open in Chrome all day.

**Comment by chid** [comment]
That sounds pretty nice, I have stopped using Pidgin since there was no skype plugin that didn't use skype.

**Comment by Tatsh** [comment]
@child Unfortunately that is not possible at the moment. Blame Skype (now Microsoft).

The best you can do is live with text only Skype within Pidgin (via the plugin) and hide Skype away (various ways to achieve this with X and Windows; not sure about OS X). I do not do this. I just keep both of them open side by side in KDE :unamused:

**Comment by akohlsmith** [comment]
The skype plugin on OSX isn't so great; it doesn't always pick up the messages that come in to Skype, and you get both the Skype app and Adium notifying you. You can shut off notifications in the Skype app but as I said, Adium doesn't always get every message Skype receives.

An Adium/Pidgin plugin for SMS would be fantastic. I'm very good with C, but a total newbie with ObjC. The only thing the two languages seem to share is the alphabetical letter. :-(

**Comment by Tatsh** [comment]
There's no need to know Objective C arguably, at least not a lot anyway. The structures might be in Apple's preferred formats, in which case things like https://github.com/libimobiledevice/libplist can be used (for parsing property list XML/binary as an example). The default language for libpurple is C. So of course what gives you the most access to internals is the C plugin API.

libpurple plugins can also be written in Python and Perl if the user has the pre-requisites. And libplist has Python bindings (I recently fixed them to work with Python 3 too).

**Comment by akohlsmith** [comment]
@tatsh I was referring more to getting WifiSMS working with iOS 7.x

**Comment by chid** [comment]
Since we're talking about RM3, there's really no need to know any ObjC/Apple formats. If you use python it will be nice to develop with the websocket module

**Comment by chid** [comment]
Receiving messages is just a chunk of json

```
[
  {
    "error_status" : 0,
    "isDelivered" : 1,
    "isSent" : false,
    "timeRead" : "11:38",
    "messageParts" : [
      {
        "text" : "YOUR MESSAGE HERE",
        "type" : "text",
        "partid" : "???"
      }
    ],
    "event" : "msgUpdated",
    "recipHashID" : "THEIR PHONE NUMBER",
    "guid" : "A2F5A53C-0CC7-AAD1-13B5-825AE50FDE55",
    "isRead" : 1,
    "particID" : "THEIR PHONE NUMBER",
    "isIM" : 0,
    "date" : 1393288721
  }
]
```

and sending messages is just a simple post request :)
